### PR TITLE
Bigtable hbase2x example for Async Operations

### PR DESCRIPTION
**Pull request by rahulKQL**
Demonstrate basic operations with asynchronous connections.

Added instruction for execution of this example

Include async samples in parent pom.xml

### CODE REVIEW

File: java/simple-async-samples/README.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
+# Cloud Bigtable Basic Async Operations
```
Comment (rahulKQL) [inline java/simple-async-samples/README.md:1]: @romlogic,

I think we can skip this file, As this is a standard template for all samples.

File: java/simple-async-samples/README.md [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+
+**Table of Contents**
+
+- [Downloading the sample](#downloading-the-sample)
+- [Costs](#costs)
+- [Before you begin](#before-you-begin)
+  - [Installing Maven](#installing-maven)
+  - [Creating a Project in the Google Cloud Platform Console](#creating-a-project-in-the-google-cloud-platform-console)
+  - [Enabling billing for your project.](#enabling-billing-for-your-project)
+  - [Install the Google Cloud SDK.](#install-the-google-cloud-sdk)
```
Comment (romlogic) [inline java/simple-async-samples/README.md:20]: Installing the Google Cloud ...

File: java/simple-async-samples/README.md [hunk 3]
```diff
@@ -0,0 +1,212 @@
...
+1. Give your project a name.
+1. Make a note of the project ID, which might be different from the project
+   name. The project ID is used in commands and in configurations.
+
+[cloud-console]: https://console.cloud.google.com/
+
+### Enabling billing for your project.
+
+If you haven't already enabled billing for your project, [enable
+billing][enable-billing] now.  Enabling billing allows is required to use Cloud Bigtable
```
Comment (romlogic) [inline java/simple-async-samples/README.md:97]: Enabling billing is required to use Cloud Bigtable...

File: java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/BasicOperations.java [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+public class BasicOperations {
+
+  private final CompletableFuture<AsyncConnection> asyncConnection;
+  private final TableName tableName;
+
+  public BasicOperations(){
+    //Creates a future for a AsyncConnection.
+    asyncConnection = ConnectionFactory.createAsyncConnection();
+
+    // Creates a future of TableName
```
Comment (romlogic) [inline java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/BasicOperations.java]: add full stop

File: java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/BasicOperations.java [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+  }
+
+  /**
+   * Deletes table present in bigtable.
+   */
+  public CompletableFuture<Void> deleteTable() {
+    return asyncConnection.thenCompose(conn -> conn.getAdmin().deleteTable(tableName));
+  }
+
+  // necessary properties to connect with bigtable.
```
Comment (romlogic) [inline java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/BasicOperations.java]: Necessary ..(capitalise N)

File: java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/BasicOperations.java [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+    System.out.println(" -------- Started Bigtable Example -------- \n");
+    BasicOperations example = new BasicOperations();
+
+    try {
+      System.out.println("\n -------- Started put and get against a single row -------- ");
+      example.putAndGet()
+          // Prints the received details with {@link Utitlity#printData(Result)}.
+          .thenAccept(Utitlity::printData)
+          //Blocks the Future<Result> to finish fetching & printing
```
Comment (romlogic) [inline java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/BasicOperations.java]: add full stop.

File: java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/BasicOperations.java [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+  }
+
+  /**
+   * Creates a row, then fetches present details of "first-row" using {@link AsyncTable#get(Get)}.
+   */
+  public CompletableFuture<Result> putAndGet() {
+    //To fetch details of a row with provided row-id.
+    Get get = new Get("first-row".getBytes());
+    return getAsyncTable().thenCompose(table ->
+        // creates data in the provided table.
```
Comment (romlogic) [inline java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/BasicOperations.java]: Creates...

File: java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/BasicOperations.java [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+        createSingleRowData(table)
+            // fetches the Result object.
```
Comment (romlogic) [inline java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/BasicOperations.java]: Fetches...

File: java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/BasicOperations.java [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+            .thenCompose(next -> table.get(get)));
+  }
+
+  /**
+   * Creates multiple row in the table, Once finished it scans with filter condition using
```
Comment (romlogic) [inline java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/BasicOperations.java]: Creates multiple row in the table; once finished, it scans with filter...

File: java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/Utitlity.java [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+  /**
+   * Returns a string, to be used as table name.
+   */
+  private static String getTableName(){
+    return "Test_Table_" + UUID.randomUUID().toString().substring(0, 10);
+  }
+
+  /**
+   * Create asynchronously a table in Bigtable instance. First checks if a table with the same
+   * name already exist. If yes then uniquely generates another tableName with UUID.
```
Comment (romlogic) [inline java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/Utitlity.java]: If yes, then ...

File: java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/Utitlity.java [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+   */
+  @VisibleForTesting
+  static CompletableFuture<TableName> createTable(AsyncAdmin admin) {
+    final TableName table = TableName.valueOf(getTableName());
+
+    // Creates columnFamily to be used throughout out rows.
```
Comment (romlogic) [inline java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/Utitlity.java]: Creates columnFamily to be used throughout out rows. (what are out rows? looks extra. )

File: java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/Utitlity.java [hunk 12]
```diff
@@ -0,0 +1,155 @@
...
+    return admin.tableExists(table).thenApply(isExist -> {
+      if (isExist) {
+        System.out.println("Table already existed, So creating with a new name");
+        return TableName.valueOf(getTableName());
+      }
+      System.out.printf("No table existed with \"%s\" name, now creating one\n", table.getNameAsString());
+      return table;
+    }).thenCompose(tableName -> {
+
+      // Adds ColumnFamilies to table
```
Comment (romlogic) [inline java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/Utitlity.java]: add full stop.

File: java/simple-async-samples/src/test/java/com/example/bigtable/simpleasync/TestBasicOperations.java [hunk 13]
```diff
@@ -0,0 +1,136 @@
...
+  public void testTableContainsFirstRow() {
+    CompletableFuture<Result> future = basicOp.putAndGet();
+    Result result = null;
+    try {
+     result = future.get();
+      System.out.println("Fetched");
+    } catch (InterruptedException | ExecutionException e) {
+      throw new AssertionError("error while fetching row details");
+    }
+    // Making sure that we received row details
```
Comment (romlogic) [inline java/simple-async-samples/src/test/java/com/example/bigtable/simpleasync/TestBasicOperations.java]: // Making sure that we received row details. or, if there is scope to rephrase all sentences, "Make sure that you receive row details."

File: java/simple-async-samples/src/test/java/com/example/bigtable/simpleasync/TestBasicOperations.java [hunk 14]
```diff
@@ -0,0 +1,136 @@
...
+  @Test
+  public void testTableForScan() {
+    CompletableFuture<List<Result>> future = basicOp.putAllAndScanAll();
+    List<Result> result = null;
+    try {
+      result = future.get();
+    } catch (InterruptedException | ExecutionException e) {
+      throw new AssertionError("error while fetching row details");
+    }
+    // Making sure that we received row details
```
Comment (romlogic) [inline java/simple-async-samples/src/test/java/com/example/bigtable/simpleasync/TestBasicOperations.java]: // Making sure that we received row details.

File: java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/Utitlity.java [hunk 15]
```diff
@@ -0,0 +1,155 @@
...
+    // After below future success, table will have two more rows.
+    return asyncTable.putAll(puts);
+  }
+
+  /**
+   * Prints data present in {@link Result} object.
+   */
+  @VisibleForTesting
+  static void printData(Result result) {
+    // Iterate of the results. Each Cell is a value for column
```
Comment (romlogic) [inline java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/Utitlity.java]: Iteration of the results. Each Cell is a value for column

File: java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/Utitlity.java [hunk 16]
```diff
@@ -0,0 +1,155 @@
...
+    // Creates a row with "second-scan-row" as row-key with different column, qualifier & value.
+    Put secondRow = new Put("second-scan-row".getBytes())
+        .addColumn(FAMILY_2, "qualifier-scan-one".getBytes(), "first-value".getBytes())
+        .addColumn(FAMILY_1, "qualifier-scan-two".getBytes(), "second-value".getBytes())
+        .addColumn(FAMILY_2, "qualifier-scan-three".getBytes(), "third value".getBytes());
+
+    // AsyncAdmin#putAll expects List<Put>, so creating an immutable list.
+    ImmutableList<Put> puts = ImmutableList.of(firstRow, secondRow);
+
+    // After below future success, table will have two more rows.
```
Comment (romlogic) [inline java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/Utitlity.java]: After its success, table ...

File: java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/Utility.java [hunk 17]
```diff
@@ -138,18 +138,17 @@ private static String getTableName() {
...
+    // After it resolves, table will have two more rows.
     return asyncTable.putAll(puts);
   }
 
   /** Prints data present in {@link Result} object. */
   @VisibleForTesting
   static void printData(Result result) {
-    // Iterate of the results. Each Cell is a value for column
-    // so multiple Cells will be processed for each row.
+    // Iterates the results. Each Cell is a value for column, So multiple Cells will be
```
Comment (romlogic) [inline java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/Utility.java:148]: should not capitalize S in the so

File: java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/BasicOperations.java [hunk 18]
```diff
@@ -45,7 +45,7 @@ public BasicOperations() {
     // Creates a future for a AsyncConnection.
```
Comment (romlogic) [inline java/simple-async-samples/src/main/java/com/example/bigtable/simpleasync/BasicOperations.java:45]: for an AsyncConnection.

**Pull request closed without merging**